docs/api/platform/Error.md

```diff
@@ -130,6 +130,37 @@ Mbed OS application and system developers may need to define error codes specifi
+
+```void mbed_error_reboot_callback(mbed_error_ctx *error_context);```
+
+Note that this callback will be invoked before the system starts executing application `main()`. So the implementation of callback should be aware any resource limitations or availability of resources which are yet to be initialized by application `main()`. Also note that the callback is invoked only when there is a new error. 
+
+#### Adding Crash-data-RAM region for crash reporting
+As mentioned above, the crash reporting feature requires a special memory region, called Crash-data-RAM to work. This region is 256 bytes in size and is allocated using linker scripts for the target for each toolchain. Although all platforms support crash reporting feature, not all targets are currently modified to allocate this Crash-data-RAM region.
+See `mbed_lib.json` in platform directory to see which targets currently enabled with crash reporting. In order to enable crash reporting in other targets, you must modify the linker scripts for those targets to allocate the Crash-data-RAM region. You may refer the linker scripts for one of the targets already enabled with crash reporting to understand how the Crash-data-RAM region is allocated. Below are some guidelines to make the linker script changes.
+
+* The region size should be 256 bytes and aligned at 8-byte offset.
+* If you are enabling the Crash-data-RAM for *ARM compiler*, linker script must export the following symbols:
+__Image$$RW_m_crash_data$$ZI$$Base__ - Indicates start address of Crash-data-RAM region.
+__Image$$RW_m_crash_data$$ZI$$Size__ - Indicates size of Crash-data-RAM region.
+* If you are enabling the Crash-data-RAM for *GCC ARM compiler* or *IAR Compiler*, linker script must export the following symbols:
+__\_\_CRASH_DATA_RAM_START\_\___ - Indicates start address of Crash-data-RAM region.
+__\_\_CRASH_DATA_RAM_END\_\___ - Indicates end address of Crash-data-RAM region.
+
+It's important that these regions should be marked with appropriate attributes(based on toolchain) to mark them as uninitialized region. For example, for ARM Compiler Crash-data-RAM can be marked with attribute *EMPTY*. There is no hard requirement about the placement of this region. The only requirement is that it should be placed such that no other entity is overwriting this region when rebooted or at runtime. But in order to avoid fragmentation its best placed just after the vector table region, or if there is no vector table region, it can be placed at the bottom of RAM(lowest address).
```

Comment:
```suggestion
It's important that these regions should be marked with appropriate attributes(based on toolchain) to mark them as uninitialized regions. For example, the ARM Compiler Crash-data-RAM region can be marked with attribute *EMPTY*. There is no hard requirement about the placement of this region. The only requirement is that it should be placed such that no other entity is overwriting this region when rebooted or at runtime. But in order to avoid fragmentation its best placed just after the vector table region, or if there is no vector table region, it can be placed at the bottom of RAM(lowest address).
```